module/platform/Webots/src/Webots.cpp

```diff
@@ -335,16 +336,16 @@ namespace module::platform {
                     emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
                     // If we have not seen the welcome message yet, look for it
                     if (!connection_active) {
-                        // Initaliase the string with ???????
-                        std::string initial_message = std::string(7, '?');
-                        const int n                 = ::read(fd, initial_message.data(), sizeof(initial_message));
+                        // Initaliase the string with 0s
+                        std::array<char, 8> initial_message{};
+                        const int n = ::read(fd, initial_message.data(), initial_message.size());
 
                         if (n >= 0) {
-                            if (initial_message == "Welcome") {
+                            if (std::strncmp(initial_message.data(), "Welcome", initial_message.size()) == 0) {
```

Comment:
Does this work instead?
```suggestion
                            if (std::string("Welcome") == initial_message.data()) {
```

Reply:
If you like you can probably swap the sides of the ==